sql/core/src/main/scala/org/apache/spark/sql/execution/joins/HashOuterJoin.scala

```diff
@@ -78,8 +78,11 @@ trait HashOuterJoin {
 
   @transient private[this] lazy val leftNullRow = new GenericInternalRow(left.output.length)
   @transient private[this] lazy val rightNullRow = new GenericInternalRow(right.output.length)
-  @transient private[this] lazy val boundCondition =
+  @transient private[this] lazy val boundCondition = if (condition.isDefined) {
```

Comment:
It's not related to this PR, just make it to aligned with others (a little faster for empty condition).